base/array.jl

```diff
@@ -420,6 +420,11 @@ julia> collect(Float64, 1:2:5)
  1.0
  3.0
  5.0
+
+julia> collect(Dict(1 => 2, 2 => "foo"))
```

Comment:
k. would be good for a different pr (and probably different collect method's docstring) but not needed now